features/0592-indy-attachments/README.md

```diff
@@ -20,51 +20,107 @@ Allows Indy-style credentials to be used with credential-related protocols that
         }
     }]
 }
 ```
 
+### cred abstract format
+
+This format is used to clarify the structure and semantics (but not the concrete data values) of a potential credential, in offers sent from issuer to potential holder.
+
+The identifier for this format is `hlindy@v2.0` in the context of an Offer Credential message. It is a base64-encoded version of the data returned from [`indy_issuer_create_credential_offer()`](https://github.com/hyperledger/indy-sdk/blob/57dcdae74164d1c7aa06f2cccecaae121cefac25/libindy/src/api/anoncreds.rs#L280).
+
+The JSON (non-base64-encoded) structure might look like this:
+
+```json
+{
+    "schema_id": "4RW6QK2HZhHxa2tg7t1jqt:2:bcgov-mines-act-permit.bcgov-mines-permitting:0.2.0",
+    "cred_def_id": "4RW6QK2HZhHxa2tg7t1jqt:3:CL:58160:default",
+    "nonce": "57a62300-fbe2-4f08-ace0-6c329c5210e1",
+    "key_correctness_proof" : <key_correctness_proof>
+}
+```
+
+A complete [`offer-credential` message from the Issue Credential protocol 2.0](../0453-issue-credential-v2/README.md#offer-credential) embeds this format at `/offers~attach/data/base64`:
+
+```json
+{
+    "@type": "https://didcomm.org/issue-credential/%VER/offer-credential",
+    "@id": "<uuid of offer message>",
+    "replacement_id": "<issuer unique id>",
+    "comment": "<some comment>",
+    "credential_preview": <json-ld object>,
+    "formats" : [
+        {
+            "attach_id" : "<attach@id value>",
+            "format": "hlindy@v2.0"
+        }
+    ],
+    "offers~attach": [
+        {
+            "@id": "<attach@id value>",
+            "mime-type": "application/json",
+            "data": {
+                "base64": "ewogICAgInNjaGVtYV9pZCI6ICI0Ulc2UUsySFpoS... (clipped)... jb3JyZWN0bmVzc19wcm9vZj4KfQ=="
+            }
+        }
+    ]
+}
+```
+
 The same structure can be embedded at `/offers~attach/data/base64` in an [`offer-credential` message](../0453-issue-credential-v2/README.md#offer-credential). 
 
 ### cred request format
 
 This format is used to formally request a credential. It differs from the credential abstract above in that it contains a cryptographic commitment to a link secret; an issuer can therefore use it to bind a concrete instance of an issued credential to the appropriate holder. (In contrast, the credential abstract describes the schema and cred def, but not enough information to actually issue to a specific holder.)
 
-The identifier for this format is: `hlindy/cred-req@v2.0` It is a base64-encoded version of the data returned from [indy_prover_create_credential_req()](https://github.com/hyperledger/indy-sdk/blob/57dcdae74164d1c7aa06f2cccecaae121cefac25/libindy/src/api/anoncreds.rs#L658).
+The identifier for this format is `hlindy@v2.0` in the context of a Request Credential message. It is a base64-encoded version of the data returned from [indy_prover_create_credential_req()](https://github.com/hyperledger/indy-sdk/blob/57dcdae74164d1c7aa06f2cccecaae121cefac25/libindy/src/api/anoncreds.rs#L658).
```

Comment:
This phrase, "in the context of a Request Credential message", manifests the difference in perspectives between this PR and the last update to 0592. I expected the identifier to be for the specific attachment format: `hlindy/cred-req@v2` rather than for the ecosystem. Thus "in the context" would be unnecessary.

I think attachment formats can be used in multiple messages (including messages in protocols that haven't been written yet), so we should not hard-contextualize them by a specific message in a specific protocol.

Reply:
So those future things can define their own format markers. It's the attachment, not the format specifier that carries data. The standard defines what to expect in the attachment by format specifier. For example, in the context of a credential request, for indy one ought to expect an (encoded) indy proof-request. I thought that was why the standard defined a registry per-message in the first place.

But of course I'm happy to describe (aries-rfc) and implement (aca-py) whatever the consensus is here.

Reply:
I don't feel like I understand your reasoning here. As you're proposing to use it, it seems to me that "hlindy@v2.0" is not a format; it is a collection of formats that are expected to work together. Evidence for this assertion is that you're using at as an identifier for data structures with significantly different structures from one another; the only thing they have in common is that they form a single coherent ecosystem (right now). What I am asking for is a format identifier that would allow a new snapshot of the ecosystem where the 2.0 version of the cred offer could be combined with the 3.0 version of the cred request. In other words, I want "2.0" to modify something "proof-req" or "cred-offer", not just "hlindy".

This is a pretty significant difference of worldview; I don't think we can merge this PR until we align one way or another.